### PR TITLE
support 32bits chainid with the latest Ledger app

### PR DESCRIPTION
**Pull request by hackmod**
Please see https://github.com/LedgerHQ/blue-app-eth/commit/8260268b0214810872dabd154b476f5bb859aac0
for some larger chainId cases, returned signature v should be recomputed at the client side.

### Screenshots

for example (ESN - chainId `31102` case + Ledger Nano S + the Ethereum app 1.1.2 ),
in this case, returned `result.v` is `0x20` or `0x1f` (1-byte truncated).
image

https://ethersocial.net/tx/0x9e4b5ae1b58720e1ee0e34d2cc0e234162c824aeadfeda51d9edfbf97dfe6299

### CODE REVIEW

**Comment by Azarielle** [comment]
[Added to Asana](https://app.asana.com/0/763966770926912/779983315837063)

File: common/libs/wallet/deterministic/ledger.ts [hunk 1]
```diff
@@ -45,17 +45,29 @@ export class LedgerWallet extends HardwareWallet {
...
-    t.v = Buffer.from([t._chainId]);
+    t.v = toBuffer(t._chainId);
     t.r = toBuffer(0);
     t.s = toBuffer(0);
 
     try {
       const ethApp = await makeApp();
       const result = await ethApp.signTransaction(this.getPath(), t.serialize().toString('hex'));
 
+      let v = result.v;
```
Comment (ConnorBryan) [inline common/libs/wallet/deterministic/ledger.ts]: @hackmod 
Hey -- this code looks a little magical. Could we add more information here for the next person that sees this? An explanation comment?

File: common/libs/wallet/deterministic/ledger.ts [hunk 2]
```diff
@@ -56,10 +56,15 @@ export class LedgerWallet extends HardwareWallet {
       let v = result.v;
       if (t._chainId > 0) {
         // EIP155 support. check/recalc signature v value.
+        // Plases see https://github.com/LedgerHQ/blue-app-eth/commit/8260268b0214810872dabd154b476f5bb859aac0
```
Comment (hackmod) [inline common/libs/wallet/deterministic/ledger.ts]: @ConnorBryan // OK. done

Comment (mkrufky) [inline common/libs/wallet/deterministic/ledger.ts]: *Please see....

**Pull request merged**